service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -262,21 +262,33 @@ WHERE child_id = ANY(:childIds)
     .filterNotNull()
     .toMap()
 
-private fun mapChildReservations(rows: List<UnitAttendanceReservations.QueryRow>, serviceTimes: Map<ChildId, DailyServiceTimes>): List<UnitAttendanceReservations.ChildReservations> {
+private fun toChildDayRows(rows: List<UnitAttendanceReservations.QueryRow>, serviceTimes: Map<ChildId, DailyServiceTimes>): List<UnitAttendanceReservations.ChildDailyRecords> {
     return rows
         .groupBy { it.child }
-        .map { (child, rowsByChild) ->
-            UnitAttendanceReservations.ChildReservations(
+        .map { (child, dailyData) ->
+            UnitAttendanceReservations.ChildDailyRecords(
                 child = child.copy(dailyServiceTimes = serviceTimes.get(child.id)),
-                dailyData = rowsByChild.associateBy(
-                    keySelector = { it.date },
-                    valueTransform = {
-                        UnitAttendanceReservations.DailyChildData(
-                            reservations = it.reservations,
-                            attendance = it.attendance,
-                            absence = it.absence
-                        )
-                    }
+                dailyData = listOfNotNull(
+                    dailyData.associateBy(
+                        keySelector = { it.date },
+                        valueTransform = {
+                            UnitAttendanceReservations.ChildRecordOfDay(
+                                reservation = it.reservations.getOrNull(0),
+                                attendance = it.attendance,
+                                absence = it.absence
+                            )
+                        }
+                    ),
+                    if (dailyData.any { it.reservations.size > 1 }) dailyData.associateBy(
+                        keySelector = { it.date },
+                        valueTransform = {
+                            UnitAttendanceReservations.ChildRecordOfDay(
+                                reservation = it.reservations.getOrNull(1),
+                                attendance = it.attendance,
```

Comment:
You'll have to similarly like with reservations use only the second attendance of the day, otherwise the same attendance is shown on both rows.

Reply:
Yes of course, thanks for catching that!

Reply:
Ok now attendances are handled in the similar way than reservations